internal/manifests/collector/container.go

```diff
@@ -43,7 +43,8 @@ func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 	}
 
 	// build container ports from service ports
-	ports := getConfigContainerPorts(logger, otelcol.Spec.Config)
+	ports, err := getConfigContainerPorts(logger, otelcol.Spec.Config)
+	logger.Error(err, "container ports config")
```

Comment:
should we propagate this up from the container? I think logging here for now is fine, maybe we propagate it in a follow up.